### PR TITLE
PMM-12293 fix platform auth.

### PR DESCRIPTION
**Pull request by BupycHuk**
PMM-12293

Link to the Feature Build: SUBMODULES-3339

If this PR adds or removes or alters one or more API endpoints, please review and add or update the relevant [API documents](https://github.com/percona/pmm/tree/main/docs/api) as well:

- [ ] API Docs updated

If this PR is related to some other PRs in this or other repositories, please provide links to those PRs:

- Links to related pull requests (optional).

### CODE REVIEW

**Review by JiriCtvrtka — APPROVED** [review]

**Comment by BupycHuk** [comment]
failing CI pipeline for managed looks so weird; I have no idea why it's happening. It definitely not related to my changes

**Review by atymchuk — APPROVED** [review]

File: managed/services/supervisord/supervisord.go [hunk 1]
```diff
@@ -799,6 +799,7 @@ command =
         cfg:default.auth.generic_oauth.api_url="{{ .PerconaSSODetails.IssuerURL }}/userinfo"
         cfg:default.auth.generic_oauth.role_attribute_path="(contains(portal_admin_orgs[*], '{{ .PerconaSSODetails.OrganizationID }}') || contains(pmm_demo_ids[*], '{{ .PMMServerID }}')) && 'Admin' || 'Viewer'"
         cfg:default.auth.generic_oauth.use_pkce="true"
+        cfg:default.auth.oauth_allow_insecure_email_lookup=true
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply managed/services/supervisord/supervisord.go:802]:
```suggestion
        cfg:default.auth.oauth_allow_insecure_email_lookup="true"
```

[the remark this answers is not in the document]
Comment (atymchuk) [reply managed/services/supervisord/supervisord.go]: Well, both are fine and both work, just inconsistent across the same file :)

**Comment by atymchuk** [comment]
> failing CI pipeline for managed looks so weird; I have no idea why it's happening. It definitely not related to my changes

We can fix it by adding a temporary workaround similar to [this one](https://github.com/percona/pmm/pull/2350/files#diff-ea97b6746d7a073889ffe8a1a556ab1fb16ea9b96a64a953b01d4931609e6f45R66) until we fix the root cause.

File: .github/workflows/managed.yml [hunk 2]
```diff
@@ -55,6 +55,9 @@ jobs:
           path: ~/go/pkg/mod
           key: ${{ runner.os }}-go-modules-${{ hashFiles('**/go.sum') }}
           restore-keys: ${{ runner.os }}-go-modules-
+          
+      - name: Mark the root directory of pmm as safe
```
Comment (atymchuk) [inline .github/workflows/managed.yml:59]: It should go after `make env-compose-up`, not before.

Comment (BupycHuk) [reply .github/workflows/managed.yml]: oh, you are right. fixed

**Review by artemgavrilov — APPROVED** [review]

**Comment by BupycHuk** [comment]
Tests were fixed in main, I don't think that we need to backport them

**Pull request merged**